app/frontend/components/locale_switch.js

```diff
@@ -0,0 +1,159 @@
+   (at your option) any later version.
+
+   Ezra Bible App is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with Ezra Bible App. See the file LICENSE.
+   If not, see <http://www.gnu.org/licenses/>. */
+
+const { html, waitUntilIdle } = require('../helpers/ezra_helper.js');
+const localeController = require('../controllers/locale_controller.js');
+const locales = localeController.getAvailableLocales();
+
+const SELECT_WIDTH = '170px'; // FIXME: magic number that works with jQuery UI
+
+const template = html`
+  <style>
+    #locale-switch-box {
+      margin-top: 4em;
+    }
+    .locale-switch-container {
+      width: 100%;
+      display: flex;
+    }
+    .locale-detect {
+      line-height: 2.1em;
+      padding: 0 0.8em;
+      display: flex;
+      align-items: center;
+      justify-content: center;
+      width: calc(100% - ${SELECT_WIDTH});
+    }
+    .locale-detect i.fas+[i18n] {
+      margin-left: 0.8em;
+    }
+    .locale-switch-container .ui-selectmenu {
+      border-bottom-right-radius: 0;
+      border-top-right-radius: 0;
+      width: auto;
+    }
+    #locale-switch-box .locale-switch-container .ui-selectmenu-menu-dropdown {
+      font-size: 1em;
+      text-align: left;
+      background-color: #deedf7;
+      width: 240px !important;
+      box-shadow: 2px 2px 3px #a0a0a088;
+    }
+    .darkmode--activated #locale-switch-box .locale-switch-container .ui-selectmenu-menu-dropdown {
+      background-color: #1e1e1e;
+    }
+  </style>
+
+  <div id="locale-switch-box" class="switch-box">
+  <div class="options-header"></div>
+    <div class="locale-switch-container">
+      <select name="locale-select" class="locale-select">
+        ${locales.map(code =>
+  `<option value="${code}">${i18nHelper.getLocaleName(code, true)}</option>`)}
+      </select>
+      <div class="fg-button locale-detect ui-state-default ui-corner-right" i18n="[title]general.detect-locale-descr">
+        <i class="fas fa-globe"></i><span i18n="general.detect-locale"></span>
+      </div>    
+    </div>
+  </div>
+  `;
+
+class LocaleSwitch extends HTMLElement {
+  constructor() {
+    super();
+
+    this.innerHTML = template.innerHTML;
+
+    this.selectEl = this.querySelector('.locale-select');
+
+    this.localize();
+
+    this.querySelector('.locale-detect').addEventListener('click', () => this.handleDetectClick());
+
+    this.setSelected(i18nHelper.getLocale());
+
+    $(this.selectEl).selectmenu({
+      appendTo: this.querySelector('.locale-switch-container'),
+      width: SELECT_WIDTH,
+      change: () => this.handleChange(),
+    });
+
+    localeController.onChangeLocale(locale => this.updateOptions(locale));
```

Comment:
subscribe to locale changes. I think from the component developer perspective this approach is self-explanatory and minimalistic

Reply:
Yes, understood!